src/App.tsx

```diff
@@ -1,24 +1,240 @@
+        .then(() => removeTodoFromState(todo))
+        .catch(() => {
+          setErrorMessage(ErrorMessage.DeleteTodo);
+
+          throw new Error(ErrorMessage.DeleteTodo);
+        })
+        .finally(() => stopProcessingTodo(todo));
+    }
+
+    return todoService.editTodo(todo.id, title)
+      .then(replaceTodoInState)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UpdateTodo);
+
+        throw new Error(ErrorMessage.UpdateTodo);
+      })
+      .finally(() => stopProcessingTodo(todo));
+  }, []);
+
+  const hasTodos = todos.length > 0 || !!tempTodo;
+  const hasCompletedTodos = todos.some(({ completed }) => completed);
+  const isAllCompleted = todos.every(({ completed }) => completed);
+  const activeTodosCount = todos.filter(({ completed }) => !completed).length;
+  const totalTodoCount = todos.length;
+
+  const filteredTodos = useMemo(() => todos.filter(({ completed }) => {
+    switch (filterValue) {
+      case Status.Active:
+        return !completed;
+
+      case Status.Completed:
+        return completed;
+
+      case Status.All:
+      default:
+        return true;
+    }
+  }), [todos, filterValue]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header>
+          {hasTodos && (
+            <ToggleAllButton
+              active={isAllCompleted}
+              onToggleAll={handleToggleAllTodos}
+            />
+          )}
+
+          <NewTodo
+            refocus={totalTodoCount}
+            onAdd={handleAddTodo}
+            onError={setErrorMessage}
+          />
+        </Header>
+
+        {hasTodos && (
+          <>
+            <TodoList>
+              {filteredTodos.map(todo => (
+                <TodoItem
+                  key={todo.id}
+                  todo={todo}
+                  onDelete={handleDeleteTodo}
+                  onToggle={handleToggleTodo}
+                  onEdit={handleEditTodo}
+                  processing={processingIds.includes(todo.id)}
+                />
+              ))}
+
+              {tempTodo && (
+                <TodoItem todo={tempTodo} processing />
+              )}
+            </TodoList>
+
+            <Footer>
+              <TodoCounter value={activeTodosCount} />
+
+              <TodoFilter
+                value={filterValue}
+                onValueChange={setFilterValue}
+              />
+
+              <ClearCompletedButton
+                active={hasCompletedTodos}
+                onClear={handleClearCompletedTodos}
+              />
```

Comment:
Why not to move this inside `Footer`?

Reply:
It is inside `Footer`. Same explanation as above.